### ISSUE TITLE
Incorrect Mason setup order causes errors when running Mason commands

### ISSUE THREAD

**Issue by Anthony-Fiddes** [nvim-lua/kickstart.nvim#453]
## Describe the bug
The import order for mason, mason-lspconfig, and lspconfig is not set. This means that sometimes when you run the `:Mason` command, you get an error like this:

![image](https://github.com/nvim-lua/kickstart.nvim/assets/11233666/1862d5ad-c185-4f2b-b1a7-d260c4e160ca)

The mason-lspconfig repo explains [here](https://github.com/williamboman/mason-lspconfig.nvim#setup) that it's important the setup() functions to be called in a specific order:

> 1. mason.nvim
> 2. mason-lspconfig.nvim
> 3. Setup servers via lspconfig

I'd be happy to open a PR with the fix! You can see how I fixed it with a simple change in my own config [here](https://github.com/Anthony-Fiddes/kickstart.nvim/commit/b15dced9a5b90184dd00ab2162567417fbf93ff7).

## To Reproduce

1. Install the latest kickstart.nvim
2. Run `nvim -c Mason`
3. If you don't see the error, close neovim and repeat steps 1 and 2. I usually see it within 5 tries.

## Desktop

- OS: pop_os!
- Terminal: Kitty

## Neovim Version

```
NVIM v0.9.1                                                                                                                                        
Build type: Release                                                                                                                                
LuaJIT 2.1.0-beta3     
```

**Comment by feoh** [comment]
Please do submit a PR!

**Comment by Anthony-Fiddes** [comment]
I'd love to! I'll make the PR tomorrow after work :D

**Comment by LorenzoBettini** [comment]
If I understand correctly, this issue is fixed by #455, isn't it?

**Comment by Anthony-Fiddes** [comment]
Yep! Thanks for checking in, I meant to close this.
